src/x86/mod.rs

```diff
@@ -1,24 +1,26 @@
 //! `x86` and `x86_64` intrinsics.
 
-pub use self::ia32::*;
-pub use self::cpuid::*;
-pub use self::xsave::*;
-
-pub use self::sse::*;
-pub use self::sse2::*;
-pub use self::sse3::*;
-pub use self::ssse3::*;
-pub use self::sse41::*;
-pub use self::sse42::*;
-pub use self::avx::*;
-pub use self::avx2::*;
-
-pub use self::abm::*;
-pub use self::bmi::*;
-pub use self::bmi2::*;
-
-#[cfg(not(feature = "intel_sde"))]
-pub use self::tbm::*;
+#[macro_use]
+mod macros;
+
+mod i386;
+pub use self::i386::*;
+
+// x86 w/o sse2
+mod i586;
+pub use self::i586::*;
+
+#[cfg(any(all(target_arch = "x86", target_feature = "sse2"),
+          target_arch = "x86_64"))]
+mod i686;
+#[cfg(any(all(target_arch = "x86", target_feature = "sse2"),
+          target_arch = "x86_64"))]
+pub use self::i686::*;
```

Comment:
Everything in the `i686` modules was previously only available in `x86_64`. The check for `i686` is `x86 + sse2`.